cumulus_library/template_sql/core_medication.sql.jinja

```diff
@@ -1,105 +1,203 @@
 CREATE TABLE core__medication AS (
-    WITH 
+    WITH
+    {# Gets medication data from inline ETL extraction.
+
+    95% of the time, this dataset is the 'correct' dataset for Cerner.
+    It may not be present in EPIC datasets. #}
+    {%- if medication_datasources.inline %}
+    mcc_nonnull AS (
+        SELECT 
+            id,
+            encounter, 
+            subject, 
+            medicationcodeableconcept 
+        FROM medicationrequest 
+        WHERE medicationcodeableconcept IS NOT NULL
+        AND encounter IS NOT NULL
+    ),
+    inline_medication AS (
+        SELECT
+            id,
+            subject.reference as "patient_id",
+            encounter.reference AS "encounter_id",
+            t.r.code,
+            t.r.display,
+            t.r.system AS code_system,
+            t.r.userselected
+        FROM mcc_nonnull,
+        unnest(medicationcodeableconcept.coding) AS t(r)
+    )
+    {%- endif -%}
+
+    {# comma handling for ctas chaining #}
+    {%- if medication_datasources.inline  and
+        (medication_datasources.by_contained_ref or 
+        medication_datasources.by_external_ref)
+    -%}
+    ,
+    {%- endif %}
+    {# This section is also Cerner specific. Sometimes medication codes are blank
+    and we can get :some: info from the contains field.
+
+    As of this writing, the code coming back via contained when code is not
+    present contains text only. In these cases, we'll cast the text
+    as a display element.
```

Comment:
Oh is that right? I'd not noticed or looked for that. No codeable concepts inside the contained resources? Rude.